### ISSUE TITLE
[dev-3.x] BearerTokenInterceptor doesn't add header to requests

### ISSUE THREAD

**Issue by audkar** [apollographql/apollo-android#3160]
**Summary**
Seams that Authorization header provided by `BearerTokenInterceptor` isn't added to actual requests.

**Version**
`com.apollographql.apollo3:apollo-runtime-kotlin:3.0.0-dev11`

**Description**
I add interceptor: 
```kotlin
ApolloClient(serverUrl)
  .withInterceptor(BearerTokenInterceptor(<...>))
```
And get not authorized exception

If I simply add `Authorization` header via default headers everything works
```kotlin
ApolloClient(
  networkTransport = ApolloHttpNetworkTransport(
    serverUrl = serverUrl,
    headers = mapOf("Authorization" to "Bearer <token>",
  ),
  //<...>
), 
```
---

My guess would be that [these headers](https://github.com/apollographql/apollo-android/blob/dev-3.x/apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo3/network/http/HttpExecutionContext.kt#L7) are not used during http request (at lease IDE doesn't find usage)

**Comment by martinbonnin** [comment]
Sorry about that, I most likely broke this while adding `HttpRequestInterceptors` (https://github.com/apollographql/apollo-android/pull/3151). Next release will introduce HTTP-level interceptors that should make the process of adding headers easier. I'll post here when it's out.

**Comment by pauldavies83** [comment]
Retested on `3.0.0-dev12` and this issue appears resolved.

There is a breaking API change - you should now init your `BearerTokenInterceptor` and pass it to the `HttpNetworkTransport` as below, as the interceptor is a "network level" interceptor rather than a "graphql level" interceptor.

```
        return ApolloClient(
            networkTransport = HttpNetworkTransport(
                serverUrl = "<server-url>",
                headers = mapOf(
                    "Accept" to "application/json",
                    "Content-Type" to "application/json"
                ),
                interceptors = listOf(
                    BearerTokenInterceptor(object : TokenProvider {
                        override suspend fun currentToken(): String {
                            return "your token"
                        }

                        override suspend fun refreshToken(previousToken: String): String {
                            return "your token"
                        }
                    })
                )
            )
        )
```

**Comment by audkar** [comment]
Fixed in `3.0.0-dev12`
